bin/pyntlm_auth/app.py

```diff
@@ -0,0 +1,220 @@
+        current = netr_Authenticator()
+        current.cred.data = [x if isinstance(x, int) else ord(x) for x in auth["credential"]]
+        current.timestamp = auth["timestamp"]
+
+        subsequent = netr_Authenticator()
+
+        challenge = binascii.unhexlify(challenge)
+        response = binascii.unhexlify(nt_response)
+
+        logon = netlogon.netr_NetworkInfo()
+        logon.challenge = [x if isinstance(x, int) else ord(x) for x in challenge]
+        logon.nt = netlogon.netr_ChallengeResponse()
+        logon.nt.data = [x if isinstance(x, int) else ord(x) for x in response]
+        logon.nt.length = len(response)
+
+        logon.identity_info = netlogon.netr_IdentityInfo()
+        logon.identity_info.domain_name.string = domain
+        logon.identity_info.account_name.string = account_username
+        logon.identity_info.workstation.string = workstation
+
+        try:
+            result = secure_channel_connection.netr_LogonSamLogonWithFlags(server_name, workstation, current,
+                                                                           subsequent,
+                                                                           logon_level, logon, validation_level,
+                                                                           netr_flags)
+            (return_auth, info, foo, bar) = result
+
+            nt_key = [x if isinstance(x, str) else hex(x)[2:] for x in info.base.key.key]
+            nt_key_str = ''.join(nt_key)
+            nt_key_str = "NT_KEY: " + nt_key_str
+            print("---- NT KEY: ", nt_key_str)
+            return nt_key_str.encode("utf-8")
+        except NTSTATUSError as e:
+            nt_error_code = e.args[0]
+            nt_error_message = e.args[1]
+
+            if nt_error_code == ntstatus.NT_STATUS_NO_SUCH_USER:
+                return nt_error_message,HTTPStatus.NOT_FOUND
+            if nt_error_code == ntstatus.NT_STATUS_WRONG_PASSWORD:
+                return nt_error_message, HTTPStatus.UNAUTHORIZED
+            if nt_error_code == ntstatus.NT_STATUS_ACCOUNT_LOCKED_OUT:  # we should stop retrying after failures, then it will probably lock the user.
+                return nt_error_message, HTTPStatus.LOCKED
+            if nt_error_code == ntstatus.NT_STATUS_LOGIN_WKSTA_RESTRICTION:
+                return nt_error_message, HTTPStatus.UNAUTHORIZED
+            if nt_error_code == ntstatus.NT_STATUS_ACCOUNT_DISABLED:
+                return nt_error_message, HTTPStatus.UNAUTHORIZED
+
+            print("---- NT Error encountered while authenticating user: ", e)
+            reconnect_id = connection_id
+            return f"NT Error: {e}", HTTPStatus.INTERNAL_SERVER_ERROR
+        except Exception as e:
+            reconnect_id = connection_id
+            print("-----------General error:",e)
+            return "Error handling request", HTTPStatus.INTERNAL_SERVER_ERROR
+
+
+app = Flask(__name__)
+app.route('/ntlm/auth', methods=['POST'])(ntlm_auth_handler)
```

Comment:
Where's the endpoint to test the connection when testing it from the PacketFence GUI?